src/Translator.php

```diff
@@ -0,0 +1,542 @@
+                $this->log("Pull request was succesfully created at $pullRequestUrl");
+                $this->pullRequestUrls[] = $pullRequestUrl;
+            } else {
+                $this->log($responseBody);
+                throw new Exception("Pull request failed - status code was $code");
+            }
+        }
+    }
+
+    private function outputPullRequestUrls(): void
+    {
+        $this->log('');
+        $this->log('The following pull-requests were created:');
+        foreach ($this->pullRequestUrls as $pullRequestUrl) {
+            $this->log($pullRequestUrl);
+        }
+        $this->log('');
+    }
+
+    private function getModuleName(string $modulePath): string
+    {
+        return json_decode(file_get_contents("$modulePath/composer.json"))->name;
+    }
+
+    private function exec(string $command, ?string $cwd = null): string
+    {
+        $this->log("Running $command");
+        $process = Process::fromShellCommandline($command, $cwd);
+        $process->run();
+        if (!$process->isSuccessful()) {
+            throw new ProcessFailedException($process);
+        }
+        return trim($process->getOutput());
+    }
+
+    private function log(string $message): void
+    {
+        echo "$message\n";
+    }
+
+    /**
+     * Gets the module lang dir
+     */
+    private function getLangDirectory(string $modulePath): string
+    {
+        $sources = $this->getTransifexSources($modulePath);
+        foreach ($sources as $source) {
+            if (preg_match('#^(?<dir>.+)\\/(?<file>[^\\/]+)\\.yml$#', $source, $matches)) {
+                return $modulePath . '/' . $matches['dir'];
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Gets the directories of the JS lang folder.
+     */
+    private function getJSLangDirectories(string $modulePath): array
+    {
+        $sources = $this->getTransifexSources($modulePath);
+        $dirs = [];
+        foreach ($sources as $source) {
+            // Strip out /src/ dir and trailing file.js
+            if (preg_match('#^(?<dir>.+)\\/src\\/(?<file>[^\\/]+)\\.js(on)?$#', $source, $matches)) {
+                $dirs[] = $modulePath . '/' . $matches['dir'];
+            }
+        }
+        return $dirs;
+    }
+
+    /**
+     * Get list of transifex source files. E.g. lang/en.yml
+     */
+    private function getTransifexSources(string $modulePath): array
+    {
+        $path = "$modulePath/.tx/config";
+        $content = file_get_contents($path);
+        $sources = [];
+        foreach (preg_split('~\R~u', $content) as $line) {
+            if (preg_match('#source_file\s=\s(?<path>\S+)#', $line, $matches)) {
+                $sources[] = $matches['path'];
+            }
+        }
+        return $sources;
+    }
+
+    /**
+     * Generate javascript for all modules
+     */
+    private function generateJavascript(): void
+    {
+        $this->log("Generating javascript locale files");
```

Comment:
```suggestion
        $this->log('Generating javascript locale files');
```
There are probably other cases where double quotes are used when they're not needed. I think I've mentioned this in another suggestion like this one but I won't hold up the PR for these... but they would be nice to tidy up.

Reply:
Updated 2 instances